### DISCUSSION TITLE
current save state file?

### DISCUSSION THREAD

**Discussion by DrewThomasson** [JonathanFly/bark#15] [Ideas]
Thinking that we should add a save state function to it where if it crashes while in the middle of creating the audio file (for instance a Google Colab runtime times out) then you'll have a save file to be able to resume from or something like that idk. You could have it at least keep a current version of the wav file that it keeps adding on to with every bit and a txt file that holds the last sentence it held so if your are making a audiobook then it'll start generating back from that line onwards

**Comment by JonathanFly** [comment]
I'm updating this tomorrow night (pull latest changes) I'll try to do that. 

For audiobooks you might want to use --prompt_file book.txt --prompt_file_separator (some special word or token that divides the book roughly in paragraphs) and then either --split_by_words or --split_by_chars to split within the paragraph. That way the voice will 'reset' every paragraph, and it also outputs separate wav files for each.

**Comment by JonathanFly** [comment]
This is somewhat in place. You have all the pieces, so I could make a script to blend them back together, I just didn't get around to it yet.
![image](https://user-images.githubusercontent.com/163408/235962604-d10df7b8-905d-4c2c-8b2f-c32ae9ae604b.png)
--hoarder_mode on the command line
